### DISCUSSION TITLE
Ligatures by default in patched fonts?

### DISCUSSION THREAD

**Discussion by noraj** [ryanoasis/nerd-fonts#1514] [Q&A]
By reading #976, I understand ligature is disabled from the font by default in Nerd patched Fonts. Am I correct?

So if I want to get the patched fonts with ligatures I have to patch them myself with https://github.com/ryanoasis/nerd-fonts#font-patcher? So `--removeligatures` is called by default for pre-packaged patched fonts?

I'm actually confused because I noticed in LibreOffice Writer, FiraCode Nerd Mono has ligatures working but has different behavior than ligatures in FiraCode Mono. Also the ligatures are not workign with FiraCode Nerd Mono in Kitty terminal while ligatures are enabled (default).

**Comment by Finii** [comment]
Sorry for the confusing situation.

> I understand ligature is disabled from the font by default in Nerd patched Fonts. Am I correct?

No. Usually ligatures are preserved - unless there are reasons to remove them (and then just a few of them).

The reason is that ligatures are not detected/used by all programs, and the programs that DO see the ligatures usually have a means to disable them. So it is more or less safe to have the ligatures in. And it is almost impossible to put the ligatures in again afterwards, but ligatures are used and featured by a lot programming-centric fonts.


Half of the truth is this:
https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts/FiraCode

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/4a3b8de9-6907-4883-abd5-9c443d9db35d)

But some ligatures are indeed dropped. They are only dropped when they result in unwanted behavior after the fonts are patched.
This is usully dropping the `fi` and related ligatures, if the ligature is only 'one width' wide - that is very confusing in programming contexts.

Which ligatures are dropped with a specific prepatched font can be seen in its `config.json` file, but there a very few:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/508c8dc4-ac82-4adb-9533-6ffcc5b0d218)

Example

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/e2d7eacd-257d-463f-a275-75ae080e7872)

`--removeligatures` does not automatically remove ligatures (and never has). Instead it relies on a config file that shows which table entries shall be removed in order to remove the ligatures.

* There was some 'general' ligature removal in place ages ago, which broke on every sourcefont update
* Even the code to remove the ligatures broke
* With #977 ligature removal has been removed
* and re-added for problematic ligatures with #1191

----

Regarding

> FiraCode Nerd Mono has ligatures working but has different behavior than ligatures in FiraCode Mono.

This should not be the case, but depending on how they write the ligatures it might be that strict `Monospace` (which Nerd Font Mono enforces) is not compatibles with all or some of the ligatures. We should have a look on that (please raise an Issue with screenshots if possible). And please fill in the complete questionnaire (which OS, client etc).

> Also the ligatures are not workign with FiraCode Nerd Mono in Kitty terminal while ligatures are enabled

I'm not sure what Kitty does and does-not. First you should check with something like LibreOffice Writer or Scribus. If it works there and not in Kitty it is almost always some limitation of Kitty. Anyhow, if you report in detail I can also investigate that.

**Reply by noraj** [reply]
Thanks for the details, I'll investigate further and open an issue if I find one.

**Reply by Paulo4lvarez** [reply]
Came here because SauceCodePro Nerd Font doesn't show ligatures even when enabled. Am I doing something wrong or it doesn't have them?

**Reply by Finii** [reply]
Which ligatures specifically are you missing?

And some clients / terminals can not display ligatures. Best is usually to use the font in something like LibreOffice Writer.
